authn/verifier_access_token.go

```diff
@@ -25,6 +29,34 @@ type AccessTokenClaims struct {
 	ServiceIdentity string `json:"serviceIdentity,omitempty"`
 }
 
+func (c ActorClaims) getTypedUID() string {
+	return fmt.Sprintf("%s:%s", c.Type, c.Identifier)
+}
+
+func (c ActorClaims) getK8sName() string {
+	if c.DisplayName != "" {
+		return c.DisplayName
+	}
+	if c.Username != "" {
+		return c.Username
+	}
+	if c.Email != "" {
+		return c.Email
+	}
+	return c.Identifier
+}
+
+func (c AccessTokenClaims) getInnermostActor() *ActorClaims {
+	currentActor := c.Actor
+	if currentActor != nil {
+		for currentActor.Actor != nil {
+			currentActor = currentActor.Actor
+		}
+	}
+
+	return currentActor
+}
+
```

Comment:
I think we can simplify things a bit if we introduced a function to get the identity actor that builds on `getInnermostActor`. It can then check the identity and return nil if it it's not of type `User`.

```suggestion

func (c AccessTokenClaims) getIdentity() *IDTokenClaims {
	actor := c.getInnermostActor()
	if actor != nil && types.IsIdentityType(actor.IDTokenClaims.Type, types.TypeUser) {
		return &actor.IDTokenClaims
	}

	return nil
}
```

Reply:
Oh nice. Are you suggesting to use `getIdentity` instead of `getInnermostActor` in the `auth_info.go` functions?